app/forms/degree_form.rb

```diff
@@ -65,7 +65,10 @@ def attributes=(attrs)
 
   def save_or_stash
     if degrees_form.trainee.draft?
-      save!
+      if save!
+        clear_relevant_invalid_apply_data if apply_invalid_data_includes_slug? && errors.empty?
```

Comment:
```suggestion
        clear_relevant_invalid_apply_data if apply_invalid_data_includes_slug?
```
Probably would not need the errors check, since if save! returns true, the errors should be empty anyway?